person.rb

```diff
@@ -20,4 +27,42 @@ def of_age?
   def can_use_services?
     is_of_age? || @parent_permission
   end
+
+  def correct_name
+    @name
+  end
 end
+
+class Decorator < Nameable
+  def initialize(nameable)
+    @nameable = nameable
+    super()
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    if @nameable.correct_name.length > 10
+      @nameable.correct_name[0..9]
+    else
+      @nameable.correct_name
+    end
+  end
+end
+
+# person = Person.new(22, 'maximilianus')
+# person.correct_name
+# capitalizedPerson = CapitalizeDecorator.new(person)
+# capitalizedPerson.correct_name
+# capitalizedTrimmedPerson = TrimmerDecorator.new(capitalizedPerson)
+# capitalizedTrimmedPerson.correct_name
```

Comment:
- [ ] [OPTIONAL] Please remove all commented code from your codebase as it pollutes it